### PR TITLE
[Microsoft.Billing] Customers resource for partner-channel

### PR DESCRIPTION
**Pull request by wilcobmsft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by NelsonDaniel** [comment]
Reassigning this b/c I will be OOF

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 1]
```diff
@@ -477,6 +477,105 @@
...
+          {
+            "name": "$filter",
+            "description": "May be used to filter using hasPermission('{permissionId}') to only return customers for which the caller has the specified permission.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$skiptoken",
+            "description": "Skiptoken is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
```
Comment (majastrz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: >nextLink [](start = 136, length = 8)

You need to add x-ms-pageable to match this as well.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 2]
```diff
@@ -477,6 +477,105 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/customers": {
```
Comment (majastrz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: >customers [](start = 71, length = 9)

Whose customers are these? (No issue with naming - just trying to understand the relationships.)

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 3]
```diff
@@ -1297,6 +1396,93 @@
...
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BillingSubscriptionsListResult"
```
Comment (majastrz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: >BillingSubscriptionsListResult [](start = 37, length = 30)

How large will this collection get? I'd recommend adding x-ms-pageable even if your service returns a single page. Much easier to add paging later.

**Comment by majastrz** [comment]
>               "$ref": "#/definitions/BillingSubscriptionsListResult"

The `BillingSubscriptionsListResult` has a next link property, but the x-ms-pageable extension is not included. Can you go through all the collection GETs to add x-ms-pageable to them?

---
Refers to: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json:1342 in e813433. [](commit_id = e813433690eec77993210718084a75ba8c3fcfea, deletion_comment = False)

**Review by majastrz — CHANGES_REQUESTED** [review]
I added a naming question and put in comments about paging all over. Can you take a look?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 3 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: Done.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 2 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: A (CSP) partner has a billing account. The customers resource lists the customers the partner does business with.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 1 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: Done.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 2 above)
Comment (majastrz) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: Gotcha. In that case calling it "customer" definitely makes sense.

---
In reply to: [295015980](https://github.com/Azure/azure-rest-api-specs/pull/6299#discussion_r295015980) [](ancestors = 295015980)

**Review by majastrz — APPROVED** [review]
Looks good.

**Comment by tjprescott** [comment]
Please resolve conflicts so the CI will be run.

**Review by yungezz — APPROVED** [review]

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/BillingSubscriptionsListByCustomer.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+              "displayName": "My subscription",
+              "subscriptionId": "6b96d3f2-9008-4a9d-912f-f87744185aa3",
+              "subscriptionBillingStatus": "Active",
+              "lastMonthCharges": {
+                "currency": "USD",
+                "value": 5000.00
+              },
+              "monthToDateCharges": {
+                "currency": "USD",
+                "value": 600
```
Comment (yungezz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/BillingSubscriptionsListByCustomer.json]: why not 600.00 as above lastMonthCharges?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 5]
```diff
@@ -750,20 +806,414 @@
...
+            "$ref": "./examples/DepartmentsListByBillingAccount.json"
           },
-          "DepartmentsListByBillingAccountNameWithExpand": {
-            "$ref": "./examples/DepartmentsListByBillingAccountNameWithExpand.json"
+          "DepartmentsListByBillingAccountWithExpand": {
+            "$ref": "./examples/DepartmentsListByBillingAccountWithExpand.json"
           }
         },
-        "operationId": "Departments_ListByBillingAccountName",
+        "operationId": "Departments_ListByBillingAccount",
```
Comment (yungezz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: this is SDK breaking change. Pls release  new service tag

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 6]
```diff
@@ -865,14 +1315,14 @@
...
+            "$ref": "./examples/EnrollmentAccountsListByBillingAccount.json"
           },
-          "EnrollmentAccountsListByBillingAccountNameWithExpand": {
-            "$ref": "./examples/EnrollmentAccountsListByBillingAccountNameWithExpand.json"
+          "EnrollmentAccountsListByBillingAccountWithExpand": {
+            "$ref": "./examples/EnrollmentAccountsListByBillingAccountWithExpand.json"
           }
         },
-        "operationId": "EnrollmentAccounts_ListByBillingAccountName",
+        "operationId": "EnrollmentAccounts_ListByBillingAccount",
```
Comment (yungezz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: operationId  change will cause  SDK breaking change

Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: Undone the change for this specific API.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 5 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: Undone the change for this specific API.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 7]
```diff
@@ -681,6 +789,9 @@
               "$ref": "#/definitions/ErrorResponse"
             }
           }
+        },
+        "x-ms-pageable": {
```
Comment (yungezz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: add 'x-ms-pagable' header to existing tag will cause sdk breaking change.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 8]
```diff
@@ -163,7 +163,7 @@
         "tags": [
           "PaymentMethods"
         ],
-        "operationId": "PaymentMethods_ListByBillingAccountName",
+        "operationId": "PaymentMethods_ListByBillingAccount",
```
[the remark this answers is not in the document]
Comment (yungezz) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: @wilcobmsft has confirmed tag 2019-10 hasn't been released in sdk. So it's ok to update operationId here. resolve this.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 7 above)
Comment (yungezz) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: confirmed with @wilcobmsft , this change is requested from ARM team, and currently no SDK released for it,  and customers using this version API is limited known one. it's safe to make the change.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
will merge this PR once CI finished.

**Pull request merged**